core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/MetadataSetupService.java

```diff
@@ -47,9 +48,9 @@
     @Autowired
     private CloudbreakEventService eventService;
 
-
     public String setupMetadata(final CloudPlatform cloudPlatform, Long stackId) throws Exception {
         Stack stack = stackService.getById(stackId);
+        MDCBuilder.buildMdcContext(stack);
         Set<CoreInstanceMetaData> coreInstanceMetaData = metadataSetups.get(cloudPlatform).collectMetadata(stack);
```

Comment:
It's worths considering to modify the interface to pass in the stack directly. Thus the MDC could be set in the caller facade (as it's done in other cases)